### PR TITLE
redis-plus-plus: added initial support

### PR DESCRIPTION
**Pull request by lpottier**
Added support for the C++ Redis client redis++
Signed-off-by: Loïc Pottier <pottier1@llnl.gov>

### CODE REVIEW

File: var/spack/repos/builtin/packages/redis-plus-plus/package.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        cxxstd = self.spec.variants["cxxstd"].value
+        use_fpic = ("+static" in self.spec) and ("+fpic" in self.spec)
+
+        args = [
+            "-DREDIS_PLUS_PLUS_CXX_STANDARD=%s" % cxxstd,
+            "-DREDIS_PLUS_PLUS_BUILD_TEST=%s" % ("+test" in self.spec),
+            "-DREDIS_PLUS_PLUS_BUILD_STATIC=%s" % ("+static" in self.spec),
+            "-DREDIS_PLUS_PLUS_BUILD_SHARED=%s" % ("+shared" in self.spec),
+            "-DREDIS_PLUS_PLUS_BUILD_STATIC_WITH_PIC=%s" % (use_fpic),
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/redis-plus-plus/package.py]: FYI. There are some helper methods inherited from `CMakePackage` that could be useful here:  https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake.

File: var/spack/repos/builtin/packages/redis-plus-plus/package.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    variant(
+        "fpic",
+        default=True,
+        description="Use Position Independent Code (-fPIC) to build the static library",
+    )
+    variant("test", default=False, description="Builds test suite")
+
+    version("1.3.6", sha256="87dcadca50c6f0403cde47eb1f79af7ac8dd5a19c3cad2bb54ba5a34f9173a3e")
+    version("1.3.5", sha256="a49a72fef26ed39d36a278fcc4e4d92822e111697b5992d8f26f70d16edc6c1f")
+    version("1.3.4", sha256="b9f2b3e0f084fe9a7360e44a9ae28aa42067fbaf027734989c778865c2d5dca5")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/redis-plus-plus/package.py:36]: Confirmed the 3 new sha256.

**Review by tldahlgren — COMMENTED** [review]
Let me know if you want to change this PR to use the helper methods or keep the code as-is.

File: var/spack/repos/builtin/packages/redis-plus-plus/package.py (hunk 1 above)
Comment (lpottier) [reply var/spack/repos/builtin/packages/redis-plus-plus/package.py]: Hi Tamara,
Thanks for the hints, I did not know that feature! I have committed a new version

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**